frontend/public/components/chargeback.tsx

```diff
@@ -256,30 +270,82 @@ class ReportData extends React.Component<ReportDataProps, ReportDataState> {
     });
   }
 
-  getColumns(data=[]) {
-    const cols = _.keys(data[0]).filter(k => {
-      if (COLS_BLACK_LIST.has(k)) {
-        return false;
-      }
-      return true;
-    });
-    return cols;
+  getColumns(data: {results: any[]}) {
+    const firstRow = _.head(data.results);
+    return _.sortBy(_.map(firstRow.values, item => item.name));
   }
 
-  getRows(data) {
-    const rows = _.map(data, (row) => {
-      return _.omitBy(row, (v, k) => {
-        return COLS_BLACK_LIST.has(k);
+  getRows(data: {results: any[]}, cols: string[]) {
+    /* Converts data to rows:
+    //  data = {
+    //   results: [
+    //     {
+    //       values: [
+    //         {
+    //           name: "period_start",
+    //           value: "2019-01-01T00:00:00Z",
+    //           unit: "date"
+    //         },
+    //         {
+    //           name: "pod_usage_cpu_core_seconds",
+    //           value: 235.73856000000004,
+    //           unit: "core_seconds"
+    //         },
+    //         ...
+    //       ]
+    //     },
+    //     {
+    //       values: [
+    //         {
+    //           name: "period_start",
+    //           value: "2019-01-01T00:00:00Z",
+    //           unit: "date"
+    //         },
+    //         {
+    //           name: "pod_usage_cpu_core_seconds",
+    //           value: 21.864000000000004,
+    //           unit: "core_seconds"
+    //
+    //   ... to ...
+    //
+    //   rows = [
+    //   {
+    //   "period_start": "2019-01-01T00:00:00Z",
+    //   "pod_usage_cpu_core_seconds": 235.73856000000004
+    //   ...
+    //  },
+    //  {
+    //   "period_start": "2019-01-01T00:00:00Z",
+    //   "pod_usage_cpu_core_seconds": 21.864000000000004
+    //   ...
+    //  },
+    //  ...
+     */
+    let rows = _.map(data.results, values => {
```

Comment:
```ts
const rowData = _.map(data.results, ({values}) => {
  return _.reduce(values, (acc, {name, value}) =>
    acc[name] = value;
    return acc;
  }, {});
});
```

But I suspect this can be simplified further  with the ordering below. Can you give an example of the final output after ordering by column? The code is hard to follow.

Reply:
> I don't see values: { values: ... } in the YAML. You're assigning the variable name on line 324.

The input is what is returned from the v2 Table api. `data.results.values[].value` etc..

I tried to implement the code you provided but it has errors, I'll work at getting past the them, but I'm a little concerned because my code does work...I'm not sure why we're looking for a different solution?

Reply:
> Can you give an example of the final output after ordering by column? The code is hard to follow.

The code normalizes the order of columns/fields within each 'row':

![image](https://user-images.githubusercontent.com/12733153/62892800-225e1780-bd17-11e9-91b5-185097e98b96.png)

Without the last bit of code/reordering, the reports look like this:
![image](https://user-images.githubusercontent.com/12733153/62892897-50435c00-bd17-11e9-9e18-2b04fe1b8838.png)

I filled: Reporting V2 Table API: 'columns' in results.values not in consistent order - https://bugzilla.redhat.com/show_bug.cgi?id=1736501 against 'Metering Operator' for this issue.

Reply:
PR to fix this is in-flight https://github.com/operator-framework/operator-metering/pull/895

Reply:
Rebased to latest `operator-framework/operator-metering`, verified @chancez fix.  Remove the column reordering logic/code.  Fixed